<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>839</width>
    <height>384</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>PyPC-NC</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,1">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,0,0,1,0">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>Status Info</string>
        </property>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="labelX">
           <property name="text">
            <string>@X</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="statusX">
           <property name="text">
            <string>04 (ready)</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="labelPx">
           <property name="text">
            <string>@Px</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="statusPx">
           <property name="text">
            <string>123</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="labelPy">
           <property name="text">
            <string>@Py</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLabel" name="statusPy">
           <property name="text">
            <string>123</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="labelPz">
           <property name="text">
            <string>@Pz</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLabel" name="statusPz">
           <property name="text">
            <string>123</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::MinimumExpanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>Workpiece Rel Loc</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_5" stretch="0,0,0,1">
         <item>
          <layout class="QFormLayout" name="formLayout_2">
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>X</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="relX">
             <property name="text">
              <string>123</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Y</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="relY">
             <property name="text">
              <string>123</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Z</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="relZ">
             <property name="text">
              <string>123</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_5">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Fixed</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <item>
              <widget class="QPushButton" name="gotoXY">
               <property name="text">
                <string>Goto XY</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="gotoXYZ">
               <property name="text">
                <string>Goto XYZ</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <widget class="QPushButton" name="storeXY">
               <property name="text">
                <string>Store XY</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="storeXYZ">
               <property name="text">
                <string>Store XYZ</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_6">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>2</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_6">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::MinimumExpanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>Manual Drive</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <layout class="QGridLayout" name="driveDirections">
           <item row="0" column="1" alignment="Qt::AlignHCenter">
            <widget class="QPushButton" name="driveYDown">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
             <property name="text">
              <string>Y-</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <spacer name="fillBelow">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>0</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="2" column="1" alignment="Qt::AlignHCenter">
            <widget class="QPushButton" name="driveYUp">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
             <property name="text">
              <string>Y+</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" alignment="Qt::AlignRight">
            <widget class="QPushButton" name="driveXDown">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="minimumSize">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
             <property name="text">
              <string>X-</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2" alignment="Qt::AlignLeft|Qt::AlignVCenter">
            <widget class="QPushButton" name="driveXUp">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
             <property name="text">
              <string>X+</string>
             </property>
            </widget>
           </item>
           <item row="0" column="4">
            <widget class="QPushButton" name="driveZUp">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
             <property name="text">
              <string>Z+</string>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Fixed</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="1" column="4">
            <widget class="QPushButton" name="driveZDown">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>32</width>
               <height>32</height>
              </size>
             </property>
             <property name="text">
              <string>Z-</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="driveDistance">
           <item>
            <widget class="QRadioButton" name="drive1Step">
             <property name="text">
              <string>1 Step</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">driveDistanceGroup</string>
             </attribute>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="drive01mm">
             <property name="text">
              <string>0.1 mm</string>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">driveDistanceGroup</string>
             </attribute>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="drive1mm">
             <property name="text">
              <string>1 mm</string>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">driveDistanceGroup</string>
             </attribute>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="drive10mm">
             <property name="text">
              <string>10 mm</string>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">driveDistanceGroup</string>
             </attribute>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="drive100mm">
             <property name="text">
              <string>100 mm</string>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">driveDistanceGroup</string>
             </attribute>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer_4">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::MinimumExpanding</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>10</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="driveSpeed">
           <item>
            <widget class="QRadioButton" name="driveFast">
             <property name="text">
              <string>fast</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">driveSpeedGroup</string>
             </attribute>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="driveSlow">
             <property name="text">
              <string>slow</string>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">driveSpeedGroup</string>
             </attribute>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>0</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QPushButton" name="stop">
          <property name="styleSheet">
           <string notr="true">background-color:red</string>
          </property>
          <property name="text">
           <string>STOP</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="importGCode">
          <property name="text">
           <string>Import G-Code</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTextEdit" name="log">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>150</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: black; color: white;</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="html">
         <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Ubuntu'; font-size:11pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Host&lt;/span&gt;:	@X&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Device&lt;/span&gt;:	@X04&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="driveDistanceGroup"/>
  <buttongroup name="driveSpeedGroup"/>
 </buttongroups>
</ui>
